osquery/core/database/database.cpp

```diff
@@ -0,0 +1,51 @@
+/**
+ *  Copyright (c) 2018-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/core/database/database.h>
+
+namespace osquery {
+
+Expected<int32_t, DatabaseError> Database::getInt32(const std::string& domain, const std::string& key) {
+  Expected<std::string, DatabaseError> string_value = getString(domain, key);
+  if (string_value) {
+    try {
+      return std::stoi(*string_value);
```

Comment:
You could use `Expected` compatible `tryTo<std::int32_t>()` here to avoid dealing with exceptions.